Sources/EdgeNetworkHandlers/NetworkResponseHandler.swift

```diff
@@ -23,6 +23,7 @@ class NetworkResponseHandler {
 
     // the order of the request events matter for matching them with the response events
     private var sentEventsWaitingResponse = ThreadSafeDictionary<String, [String]>()
+    static var currentPrivacyStatus: PrivacyStatus = EdgeConstants.DEFAULT_PRIVACY_STATUS
```

Comment:
Is it possible that `currentPrivacyStatus` could be written and read at the same time? Given that it is set from Edge.extension and read from within NetworkResponseHandler. handleStoreEventHandle.

Reply:
One suggestion would be to define a `getPrivacyStatus` function in Edge, then inject it into the NetworkResponseHandler.

```swift
// Edge.swift
init() {
   // inject closure
   networkResponseHandler = NetworkResponseHandler(getPrivacyStatus: getPrivacyStatus)
}

func getPrivacyStatus() -> PrivacyStatus {
   // read and return privacy status from shared state
}
```

```swift
// NetworkResponseHandler.swift
init(getPrivacyStatus: () -> PrivacyStatus) {
   // store the closure, then use where needed
}
```

Then you can use `getPrivacyStatus()` in `NetworkResponseHandler. handleStoreEventHandle` to read the privacy status without worry about threading concerns on the static variable. This would also increase testability as we no longer rely on a static variable being used within an instance.

The above might not compile 100%, but that's the idea I have.

Reply:
Updated, see below.